### ISSUE TITLE
various API calls on different pages encounter datastore-service exception due to missing PVRecord constructor

### ISSUE THREAD

**Issue by craigmcchesney** [craigmcchesney/datastore-web-app#42]

**Comment by craigmcchesney** [comment]
This is resolved by https://github.com/osprey-dcs/datastore-service/issues/7

**Comment by craigmcchesney** [comment]
There is a new issue in the web application after integrating the new API features, where the PvListPage encounters an exception calling the listPVs api.  On the datastore-service console there is an exception about a missing default constructor for the PVRecord class:

Failed to instantiate com.ospreydcs.datastore.model.PVRecord using constructor NO_CONSTRUCTOR with arguments

There is no stacktrace or other diagnostics on the console because the exception is caught at the point the above message is generated.  There is a similar exception thrown when calling the listSnapshotData api, with a strack trace that contains additional details (I snipped out many framework level methods between the ResultsSupplier and actual error instantiating PVRecord):

org.springframework.data.mapping.model.MappingInstantiationException: Failed to instantiate com.ospreydcs.datastore.model.PVRecord using constructor NO_CONSTRUCTOR with arguments 
	at org.springframework.data.mapping.model.ReflectionEntityInstantiator.instantiateClass(ReflectionEntityInstantiator.java:116) ~[spring-data-commons-2.7.2.jar:2.7.2]
	at 
<SNIP>
com.ospreydcs.datastore.service.query.ResultsSupplier.filterRawResults(ResultsSupplier.java:375) ~[classes/:na]
	at com.ospreydcs.datastore.service.query.ResultsSupplier.lambda$16(ResultsSupplier.java:432) ~[classes/:na]
	at com.ospreydcs.datastore.service.manager.AnnotationManager.loadAnnotation(AnnotationManager.java:81) ~[classes/:na]
	at com.ospreydcs.datastore.service.query.ResultsSupplier.get(ResultsSupplier.java:430) ~[classes/:na]
	at com.ospreydcs.datastore.service.query.QueryServiceGRPC.listSnapshotData(QueryServiceGRPC.java:331) ~[classes/:na]

I entered datastore-service issue #7 to resolve this issue.  See details for that task above.
